### ISSUE TITLE
PHP parser dependency makes the GitHub Actions fail

### ISSUE THREAD

**Issue by kastriotkastrati** [prettier/plugin-php#1822]
Hi there,

plugin-php depends on php-parser, which is locked to only git+ssh protocol rather than git+https+ssh which fails builds even though php-parser is a public repo.

best 

```
Run npm ci
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/glayzzle/php-parser.git
npm ERR! 
npm ERR! Warning: Permanently added the RSA host key for IP address '140.82.112.4' to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128
```

**Comment by Ranchman42** [comment]
I am seeing this as well on Amazon Linux 2. But with https.

```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.2 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm verb notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
npm verb notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
npm verb notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   linux
npm verb notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64

npm verb stack Error: spawn git ENOENT
npm verb stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
npm verb stack     at onErrorNT (internal/child_process.js:467:16)
npm verb stack     at processTicksAndRejections (internal/process/task_queues.js:82:21)
npm verb cwd /var/www/prod/api
npm verb Linux 4.14.231-173.361.amzn2.x86_64
npm verb argv "/usr/bin/node" "/usr/bin/npm" "i" "--verbose"
npm verb node v14.17.0
npm verb npm  v6.14.13
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -2
npm ERR! enoent Error while executing:
npm ERR! enoent undefined ls-remote -h -t https://github.com/glayzzle/php-parser.git
npm ERR! enoent
npm ERR! enoent
npm ERR! enoent spawn git ENOENT
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
npm verb exit [ -2, true ]
npm timing npm Completed in 4992ms
```

**Comment by nemchik** [comment]
Likely related https://github.com/prettier/plugin-php/issues/1813

**Comment by hc-12** [comment]
As mentioned in #1813 , upgrading npm to latest, remove package-lock.json then reinstall seem to do the trick

**Comment by xieyuheng** [comment]
The same issue on vercel.com:

```
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/glayzzle/php-parser.git
npm ERR!
npm ERR! Host key verification failed.
npm ERR! fatal: Could not read from remote repository.
npm ERR!
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR!
npm ERR! exited with error code: 128
```

What is the reason for using `php-parser` as a git dependency?

(Depending on the commit `27abcb2337ac6450c068ef064982dfabf77916a5`)

**Comment by xieyuheng** [comment]
I see, the last release of `php-parser` is "Apr 24, 2020" which is really behind the development.

Maybe we can bag upstream for a new release?

**Comment by alexander-akait** [comment]
I think we need new maintainers here, @czosel is busy, I unfortunately don't have time for that.

**Comment by czosel** [comment]
@alexander-akait the real blocker is this the parser, since we haven't heard from @ichiriac in a while. More precisely, this MR needs to be merged: https://github.com/glayzzle/php-parser/pull/809
Could you review it? Then I'd merge it into master and release it under a temporary name, e.g. `php-parser-8` on npm.

**Comment by czosel** [comment]
Should be fixed in `v0.18.0` :tada:
